src/list.c

```diff
@@ -202,6 +202,22 @@ struct SList *slist_sort(struct SList *head, bool (*before)(const void *a, const
 	return sorted;
 }
 
+void slist_move(struct SList **to, struct SList **from, bool (*predicate)(const void *val, const void *data), const void *data) {
+	if (!to || !from || !predicate)
+		return;
+
+	struct SList *f = *from;
+	while (f) {
```

Comment:
nit: Could be simplified to:
```
	struct SList *f = *from;
	while (f) {
		if (predicate(f->val, data)) {
			slist_append(to, f->val);
			slist_remove(from, &f);
		}
		f = f->nex;
	}
```

Reply:
That would indeed be much nicer.

Unfortunately not; the list is very primitive and destroys f when removing it.